dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -538,6 +547,50 @@ void MPPTask::runImpl()
     unregisterTask();
 }
 
+void MPPTask::reportStatus(const String & err_msg)
+{
+    if (!ReportStatusToCoordinator(meta.mpp_version(), meta.coordinator_address()))
+        return;
+
+    try
+    {
+        std::shared_ptr<mpp::ReportTaskStatusRequest> req = std::make_shared<mpp::ReportTaskStatusRequest>();
+        mpp::TaskMeta * req_meta = req->mutable_meta();
+        req_meta->CopyFrom(meta);
+
+        if (dag_context->collect_execution_summaries)
+        {
+            tipb::TiFlashExecutionInfo execution_info = mpp_task_statistics.genTiFlashExecutionInfo();
+            if unlikely (!execution_info.SerializeToString(req->mutable_data()))
+            {
+                LOG_ERROR(log, "Failed to serialize TiFlash execution info");
+                return;
```

Comment:
Maybe it shouldn't return here?
Because the following steps such as err_msg may not fail

Reply:
The err_msg path is a better-do mechnism, so when it failed, something unexpected happened, no err msg returned seems ok.

Reply:
One question: Will the tidb side wait for all tiflash mpptask report status?

Reply:
Not for all queries, for query with limit operator, it will.

Reply:
If directly return here, will it cause tidb to wait forever for the limit task to report the status?

Reply:
No, tidb will wait no more than a statically set thres time.

Reply:
ok